js/src/ipc/message.ts

```diff
@@ -40,6 +40,9 @@ export class MessageReader implements IterableIterator<Message> {
     public next(): IteratorResult<Message> {
         let r;
         if ((r = this.readMetadataLength()).done) { return ITERATOR_DONE; }
+        // ARROW-6313: If the first 4 bytes are continuation message, read the next 4 for the length
```

Comment:
nit: maybe add something like "otherwise assume this is a pre-0.15 file/stream with no continuation message"

Reply:
:+1: